src/flow/src/batching_mode/frontend_client.rs

```diff
@@ -133,16 +186,114 @@ impl FrontendClient {
             }
             .fail()?
         };
-        let client = client_from_urls(vec![peer.addr.clone()]);
-        let database = Database::new(DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME, client);
+        let client = Client::with_manager_and_urls(chnl_mgr.clone(), vec![peer.addr.clone()]);
+        let database = Database::new(catalog, schema, client);
         Ok(DatabaseWithPeer::new(database, peer))
     }
 
-    /// Get a database client, and possibly update it before returning.
-    pub async fn get_database_client(&self) -> Result<DatabaseWithPeer, Error> {
+    pub async fn create(
+        &self,
+        create: CreateTableExpr,
+        catalog: &str,
+        schema: &str,
+    ) -> Result<u32, Error> {
+        self.handle(
+            Request::Ddl(api::v1::DdlRequest {
+                expr: Some(api::v1::ddl_request::Expr::CreateTable(create)),
+            }),
+            catalog,
+            schema,
+            &mut None,
+        )
+        .await
+    }
+
+    /// Handle a request to frontend
+    pub(crate) async fn handle(
+        &self,
+        req: api::v1::greptime_request::Request,
+        catalog: &str,
+        schema: &str,
+        peer_desc: &mut Option<PeerDesc>,
```

Comment:
Perhaps we can move it to the method's return value.

Reply:
> Perhaps we can move it to the method's return value.

need peer_desc regardless if `handle` failed or not, but return a `(Result<u32, Error>, PeerDesc)` make `?` not work so this is probably cleaner? And this `peer_desc` is used in logging so it's fine

Reply:
Got you